pkg/webservice/handlers.go

```diff
@@ -619,6 +654,55 @@ func getQueueApplications(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPartitionApplicationsByState(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+	vars := httprouter.ParamsFromContext(r.Context())
+	if vars == nil {
+		buildJSONErrorResponse(w, MissingParamsName, http.StatusBadRequest)
+		return
+	}
+	partition := vars.ByName("partition")
+	appState := vars.ByName("state")
+
+	partitionContext := schedulerContext.GetPartitionWithoutClusterID(partition)
+	if partitionContext == nil {
+		buildJSONErrorResponse(w, PartitionDoesNotExists, http.StatusNotFound)
+		return
+	}
+	if !allowedAppStates[appState] {
+		buildJSONErrorResponse(w, allowedStatesMsg, http.StatusBadRequest)
+		return
+	}
+	var appList []*objects.Application
+	switch appState {
+	case ActiveState:
```

Comment:
We don't have "default" branch. Shouldn't we have one and return with an error message (ie. "unsupported state")?